### PR TITLE
Complete the typing of the qcodes station 

### PR DESCRIPTION
**Pull request by jenshnielsen**
@QCoDeS/core This should finaly be ready for review. 

I did not see a clean way to deprecate the access of one specific key in the `station.components` dict so this comes with a breaking change that `station.components[`config`]` is no longer there. Let me know if you have other suggestions

### CODE REVIEW

File: qcodes/station.py [hunk 1]
```diff
@@ -118,16 +123,16 @@ def __init__(self, *components: Metadatable,
         if default:
             Station.default = self
 
-        self.components: Dict[str, Metadatable] = {}
+        self.components: Dict[str, Union[Metadatable, 'StationConfig']] = {}
```
Comment (astafan8) [inline qcodes/station.py]:
```suggestion
        self.components: Dict[str, Union[Metadatable, StationConfig]] = {}
```

File: qcodes/station.py [hunk 2]
```diff
@@ -76,6 +77,10 @@ class ValidationWarning(Warning):
     pass
 
 
+class StationConfig(UserDict):
```
Comment (astafan8) [inline qcodes/station.py]: I'd vote for making first a quick PR that fixes this (or just in this PR), and then we won't need to have the ugly union in the station.components
```suggestion
class StationConfig(Metadatable):
```

Comment (jenshnielsen) [reply qcodes/station.py]: @Dominik-Vogel Do you have any opinion on this? I am happy to make the change but I may be missing the context

Comment (jenshnielsen) [reply qcodes/station.py]: I would more lean towards removing config from components and having it as a specific separate entry in the station and update snapshot_base to incorporate it

Comment (astafan8) [reply qcodes/station.py]: > I would more lean towards removing config from components and having it as a specific separate entry in the station and update snapshot_base to incorporate it

agree, let's not pollute "components".

Comment (Dominik-Vogel) [reply qcodes/station.py]: I am also all for that suggestion.

**Pull request merged**